extensions/amp-loader/0.1/amp-loader.js

```diff
@@ -257,14 +264,52 @@ class LoaderBuilder {
     `;
   }
 
+  /**
+   * `<amp-ad>s` have several different classes, so putting the code here for
+   * now since it is the safest way to make sure that they all get the correct
+   * loader.
+   *
+   * Since the implementation may have a delay before loading, we would need to
+   * make sure the ads loader is present, even if the implementation has not
+   * yet downloaded.
+   *
+   * TODO(sparhami) Move this out of amp-loader into something common for ads.
+   * @return {!Element}
+   * @private
+   */
+  getAdsLogo_() {
+    const html = htmlFor(this.element_);
+    return html`
+      <svg viewBox="0 0 72 72">
+        <rect
+          fill="none"
+          stroke="currentColor"
+          x="26.5"
+          y="29.5"
+          width="19"
+          height="13"
+          rx="2"
+        ></rect>
+        <text
+          fill="currentColor"
+          class="i-amphtml-ad-badge-label"
+          transform="translate(29.2812 40.167)"
+        >
+          Ad
+        </text>
+      </svg>
+    `;
+  }
+
   /**
    * Whether the element is an Ad.
    * @private
    * @return {boolean}
    */
   isAd_() {
-    // Not Implemented
-    return false;
+    return (
+      this.element_.tagName == 'AMP-AD' || this.element_.tagName == 'AMP-EMBED'
```

Comment:
Current behavior: we don't apply placeholder to `<amp-embed>`, because `<amp-embed>` component is usually pretty long, multiple viewport height. And we figure it's not good UI to apply loader in such case. 
https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/0.1/amp-ad-ui.js#L143

But I know @lannka propose that we remove all differences between `<amp-ad>` and `<amp-embed>` and make them exactly the same component. 

Can we please keep the same behavior for now (not adding loader to `<amp-embed>`. I'm open to future change if we agree on the loader UI for elements that's longer than 100vh. Thanks.

Reply:
After taking a look, I noticed that `<amp-embed>` currently gets the 3 dot loader (and not the ads loader). The loader element is due to the inclusion in this list:

https://github.com/ampproject/amphtml/blob/ff8827b5195577a68014e96ba1ed00c20eec9322/src/layout.js#L94-L109

which was added in https://github.com/ampproject/amphtml/commit/919ba58e4df75e49224cfbf5bf1f41dbe6cac0f8

Since we want to remove the legacy loader, I think we should either:

1. Move forward with the new ads loader for `<amp-embed>`
1. Move forward with the new generic loader for `<amp-embed>`
1. Remove the the loader element from `<amp-embed>` as part of the experiment / ramp up to measure the impact.

Any thoughts on which would be preferred?

@zhouyx @nainar @lannka

Reply:
Per our discussion, we will go with the generic loader for now, since that matches the current intent / implementation.